Description: 
 Make default charset used configurable, set to UTF-8, see:
 http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=322552
--- a/imageindex
+++ b/imageindex
@@ -290,6 +290,10 @@
 $thumbxmetatag    = 'ThumbnailX';
 $thumbymetatag    = 'ThumbnailY';
 
+# which charset to use
+#$file_charset     = 'ISO-8859-1';
+$file_charset     = 'UTF-8';
+
 # Any of the above can be overridden in an rc file in the user's home dir
 $rcfile = "$ENV{'HOME'}/.imageindexrc";
 
@@ -1430,7 +1434,7 @@
     print "<HEAD>\n";
     $verstring = &versionstring();
     printf ("<META NAME=\"GENERATOR\" CONTENT=\"imageindex %s\">\n", $verstring);
-    printf ("<META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=ISO-8859-1\">\n");
+    printf ("<META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=${file_charset}\">\n");
     print "<TITLE>$current_titletext - $filename</TITLE>\n";
     print "<LINK TYPE=\"text/css\" REL=\"stylesheet\" HREF=\"../$stylefile\">\n";
     print "</HEAD>\n<BODY>\n";
@@ -2276,7 +2280,7 @@
     print "<HEAD>\n";
     $verstring = &versionstring();
     printf ("<META NAME=\"GENERATOR\" CONTENT=\"imageindex %s\">\n", $verstring);
-    printf ("<META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=ISO-8859-1\">\n");
+    printf ("<META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=${file_charset}\">\n");
     if (defined ($write_meta_tag{$titlemetatag})) {
 	print "<META NAME=\"$titlemetatag\" CONTENT=\"$current_titletext\">\n";
     }
@@ -2628,7 +2632,7 @@
     print "<HEAD>\n";
     $verstring = &versionstring();
     printf ("<META NAME=\"GENERATOR\" CONTENT=\"imageindex %s\">\n", $verstring);
-    printf ("<META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=ISO-8859-1\">\n");
+    printf ("<META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=${file_charset}\">\n");
     print "<LINK TYPE=\"text/css\" REL=\"stylesheet\" HREF=\"../$stylefile\">";
     print "<TITLE>$current_titletext</TITLE>\n";
     print "<LINK TYPE=\"text/css\" REL=\"stylesheet\" HREF=\"../$stylefile\">\n";
@@ -2658,7 +2662,7 @@
     print "<HEAD>\n";
     $verstring = &versionstring();
     printf ("<META NAME=\"GENERATOR\" CONTENT=\"imageindex %s\">\n", $verstring);
-    printf ("<META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=ISO-8859-1\">\n");
+    printf ("<META HTTP-EQUIV=\"Content-Type\" CONTENT=\"text/html; charset=${file_charset}\">\n");
     print "<LINK TYPE=\"text/css\" REL=\"stylesheet\" HREF=\"../$stylefile\">\n";
     print "<TITLE>$current_titletext</TITLE>\n";
     print "</HEAD>\n<BODY>\n";
